src/commands/createStaticWebApp/postCreateStaticWebApp.ts

```diff
@@ -6,67 +6,75 @@
-        const maxTime: number = Date.now() + 30 * 1000; // it can take a second for the action to queue, wait 30 seconds
+export async function postCreateStaticWebApp(swaNode: StaticWebAppTreeItem): Promise<void> {
+    return await callWithTelemetryAndErrorHandling('staticWebApps.postCreateStaticWebApp', async (context: IActionContext): Promise<void> => {
+        const productionEnv: EnvironmentTreeItem | undefined = <EnvironmentTreeItem | undefined>(await swaNode.loadAllChildren(context)).find((ti) => { return ti instanceof EnvironmentTreeItem && ti.label === productionEnvironmentName; });
+        if (productionEnv) {
+            const octokitClient: Octokit = await createOctokitClient();
+            const { owner, name } = getRepoFullname(productionEnv.repositoryUrl);
+            let deployActionNode: ActionTreeItem | undefined;
+            const maxTime: number = Date.now() + 30 * 1000; // it can take a little for the action to queue in GitHub, wait for 30 seconds
 
-        while (!deployActionNode) {
-            await productionEnv.actionsTreeItem.refresh();
-            const actions: ActionTreeItem[] = <ActionTreeItem[]>(await productionEnv.actionsTreeItem.loadAllChildren(context));
-            deployActionNode = actions.find(async (ti) => {
-                const workflow: ActionsGetWorkflowResponseData = (await octokitClient.actions.getWorkflow({ owner, repo: name, workflow_id: ti.data.workflow_id })).data;
-                return workflow.path.includes(swaNode.defaultHostname) && ensureStatus(ti.data) === Status.InProgress;
-            });
+            while (!deployActionNode) {
+                await productionEnv.actionsTreeItem.refresh();
+                const actionTreeItems: ActionTreeItem[] = <ActionTreeItem[]>(await productionEnv.actionsTreeItem.loadAllChildren(context));
+                const filteredTreeItems: ActionTreeItem[] = actionTreeItems.filter(ti => { return ti.data.status !== Status.Completed; }); // only looking at on-going or queued jobs
 
-            if (Date.now() > maxTime) {
-                throw new UserCancelledError();
+                const promises: Promise<void>[] = filteredTreeItems.map(async ti => {
+                    const workflow: ActionsGetWorkflowResponseData = (await octokitClient.actions.getWorkflow({ owner, repo: name, workflow_id: ti.data.workflow_id })).data;
+                    // example of defaultHostname: 'black-bay-07228711e.azurestaticapps.net'
+                    // example of workflow path: '.github/workflows/azure-static-web-apps-black-bay-07228711e.yml'
+                    // to verify the workflow matches the current default host, we need to remove the 'azurestaticapps.net' portions of the defaultHostname
+                    if (workflow.path.includes(swaNode.defaultHostname.split('.')[0])) {
+                        deployActionNode = ti;
+                    }
+                });
+                await Promise.all(promises);
```

Comment:
Could you explain why you await the promises here? I'm not for sure but it looks like `filteredTreeItems.map` will generate an array of all void values since nothing is returned from within the map call

Reply:
This is so the API call `getWorkflow` gets called in parallel.  If I iterate through it in a for loop, it'll be done synchronously (since I'm awaiting each one, one at a time).

The promise isn't returning anything, but it's setting the `deployActionNode` which is what signals the end of the while loop.  What I'm doing here is getting the workflow file associated to the action because it tells me what .yml file it's using to configure, which should match the defaultHostname of the static web app.

Reply:
That's pretty nifty. I think it might be worth adding a short comment like "Run `getWorkflow` in parallel" to help less advanced typescript peeps like myself understand 😄